### PR TITLE
Try to fix performance test failure on trunk

### PR DESCRIPTION
**Pull request by t-hamano**
## What?
This PR fixes a problem in trunk where [performance tests compared with base branch](https://github.com/WordPress/gutenberg/actions/workflows/performance.yml?query=branch%3Atrunk) fail.

## Why?
The GitHub Action log shows:

```
FAIL packages/e2e-tests/specs/performance/site-editor.test.js (62.334 s)
  Site Editor Performance
    ✓ Loading (1 of 4) (2134 ms)
    ✓ Loading (2 of 4) (2112 ms)
    ✓ Loading (3 of 4) (2469 ms)
    ✓ Loading (4 of 4) (13246 ms)
    ✕ Typing (18213 ms)

  ● Site Editor Performance › Typing

    Node is detached from document

      152 |
      153 | 		// Insert a new paragraph right under the first one.
    > 154 | 		await firstParagraph.click(); // Once to select the block overlay.
          | 		^
      155 | 		await firstParagraph.click(); // Once again to select the paragraph.
      1[56](https://github.com/WordPress/gutenberg/actions/runs/5235189272/jobs/9451845697#step:6:57) | 		await insertBlock( 'Paragraph' );
      1[57](https://github.com/WordPress/gutenberg/actions/runs/5235189272/jobs/9451845697#step:6:58) |

      at ElementHandle._scrollIntoViewIfNeeded (../../node_modules/puppeteer-core/src/common/JSHandle.ts:475:22)
          at runMicrotasks (<anonymous>)
      at ElementHandle.click (../../node_modules/puppeteer-core/src/common/JSHandle.ts:[61](https://github.com/WordPress/gutenberg/actions/runs/5235189272/jobs/9451845697#step:6:62)7:5)
      at Object.<anonymous> (specs/performance/site-editor.test.js:154:3)
```

The code indicated by this error is:

https://github.com/WordPress/gutenberg/blob/1746e8c1feff5252b6235442a1c3172c8f1c658c/packages/e2e-tests/specs/performance/site-editor.test.js#L145-L156

From the Artifacts snapshot, it appears that the site editor is successfully switching to edit mode, but is failing to click on the paragraph block:

Typing 2023-06-11T12-03-17

I expected that when switching to edit mode, the re-rendering of the DOM would have caused the previously defined `firstParagraph` to be detached ([one source](https://github.com/puppeteer/puppeteer/issues/3496#issuecomment-436462052)). I do not know the exact cause of this problem, but one factor might be that the commit in the base branch was changed by #51381.

## How?
After switching to edit mode, I had puppeteer look for the first paragraph block again. In addition, the if statement is _temporarily_ changed in order to run the performance test compared with base branch in PR as well.

## Testing Instructions

In the GitHub Actions performed on this PR, confirm that the performance test compared with base branch is successful.

### CODE REVIEW

**Comment by youknowriad** [comment]
> I expected that when switching to edit mode, the re-rendering of the DOM would have caused the previously defined firstParagraph to be detached (https://github.com/puppeteer/puppeteer/issues/3496#issuecomment-436462052).

I actually think that's a problem in itself, maybe we're remounting the components when entering edit mode and that's something I wanted to avoid initially to have a smooth experience (for example embeds/videos don't reload). In other words, why do we have a new DOM node. I think it might a regression in some PR that was unintentional.

**Comment by t-hamano** [comment]
> maybe we're remounting the components when entering edit mode and that's something I wanted to avoid initially to have a smooth experience (for example embeds/videos don't reload).

Perhaps the re-rendering of the DOM is not the underlying problem. As shown below, the video plays seamlessly when switching modes (I understand this to mean that the DOM is not being re-rendered). So it is unclear why Node is detached...

https://github.com/WordPress/gutenberg/assets/54422211/8fc67219-7f4f-4a07-94cf-defededd8efe

**Comment by kevin940726** [comment]
FWIW this was also happening in a lot of playwright tests before. We [fixed](https://github.com/WordPress/gutenberg/pull/49374) it by simply waiting for the iframe to be ready instead of jumping into the rabbit hole. Possibly the fix could serve as an inspiration here?

**Comment by WunderBart** [comment]
> > maybe we're remounting the components when entering edit mode and that's something I wanted to avoid initially to have a smooth experience (for example embeds/videos don't reload).
> 
> Perhaps the re-rendering of the DOM is not the underlying problem. As shown below, the video plays seamlessly when switching modes (I understand this to mean that the DOM is not being re-rendered). So it is unclear why Node is detached...
> 
>  3f5dc38713a9413345a0ecc014e8f3f3.mp4

While migrating to Playwright, [I also noticed](https://github.com/WordPress/gutenberg/pull/51084/files#diff-c706a137637e6ef100f4e3e20b74441375979738f8a8bc5ea8e9cc07b5df689cR162-R175) that one more click is now needed to get to the target paragraph. That extra layer seems to be the `Post Content` block which **needs to be explicitly clicked** because (at least Playwright) won't click the underlying paragraph until it's not covered. Clicking the `Post Content` after entering Edit Mode did the trick, and the test seems stable on my branch. Maybe it will also help with this one?

The [click-path to the paragraph](https://github.com/WordPress/gutenberg/pull/51084/files#diff-c706a137637e6ef100f4e3e20b74441375979738f8a8bc5ea8e9cc07b5df689cR162-R175) that worked for me was:
1. `body` (to enter Edit Mode)
2. `Post Content` (to enter post content)
3. `Paragraph` (to enter paragraph).

**Comment by youknowriad** [comment]
I think I understand more now. It seems previously it didn't matter if the paragraph block was actually selected or not, we were just adding a "paragraph" block to the template. But since now the editor is by default in "page mode" (no edits/insertion in the template) we can't really insert blocks in the template. So this is probably a side effect of that PR cc @noisysocks 

The question now, is what approach should we do to fix this, we have two options:

 - Insert paragraphs within "post content" (two clicks).
 - Switch to template mode and insert paragraphs within the template.

If we don't want to change the purpose of the metric itself, we should probably do the second but the first one is fine. But it means that potentially we can't compare the graph results with previous numbers of the metric.

**Comment by t-hamano** [comment]
I may be missing something, but from the beginning when the typing performance test was added, the codebase seems to indicate that typing is intended to take place within the post content in page mode 🤔

https://github.com/WordPress/gutenberg/pull/32553/files?diff=unified&w=0#diff-dac6b5792388c0d2844f5ec7201465c2a1d35b2bc18c580e0aa811a812af2f17R74-R93

**Comment by youknowriad** [comment]
What was happening (at least for some time now) is that when we clicked the paragraph, it was the "post content" block that got selected and not the paragraph meaning that "inserting a paragraph" would insert it after the post content block, not within.

**Comment by t-hamano** [comment]
Okay, I think I may have understood. The base branch commit being compared in the performance test was changed at #51381, but it certainly looks like both of these codes just selected the post content block.

Before: debd225d007f4e441ceec80fbd6fa96653f94737

https://github.com/WordPress/gutenberg/blob/debd225d007f4e441ceec80fbd6fa96653f94737/packages/e2e-tests/specs/performance/site-editor.test.js#L115-L122

After: 843a3053aca918bb10b939be28e676f8e71b751b

https://github.com/WordPress/gutenberg/blob/843a3053aca918bb10b939be28e676f8e71b751b/packages/e2e-tests/specs/performance/site-editor.test.js#L118-L126

**Review by youknowriad — APPROVED** [review]
I'm going to approve this PR. I think both options are valid (either add a paragraph within post content or outside post content), we can accept that the metric values change a bit.

File: .github/workflows/performance.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ jobs:
                   ./bin/plugin/cli.js perf "wp/$WP_MAJOR" "$PREVIOUS_RELEASE_BRANCH" "$CURRENT_RELEASE_BRANCH" --tests-branch $GITHUB_SHA --wp-version "$WP_MAJOR"
 
             - name: Compare performance with base branch
-              if: github.event_name == 'push'
+              if: github.event_name == 'pull_request'
```
Comment (youknowriad) [inline .github/workflows/performance.yml]: This should be reverted right?

Comment (t-hamano) [reply .github/workflows/performance.yml]: Yes, this change was made temporarily to allow pull requests to also run the performance test compared with base branch. Unnecessary changes were reverted in 467f513.

Comment (youknowriad) [reply .github/workflows/performance.yml]: Just noting that changing the event name like that in PRs had some unwanted side effects, basically all the temporary commits in this PR started logging numbers in codevitals.run 

I think we should stop doing that in the future. (at least for the part that calls the log script)

Comment (kevin940726) [reply .github/workflows/performance.yml]: Do we have some sort of permission check as GH Actions does for `pull_request` and `pull_request_target`? Does this mean that anyone opening a PR could potentially skew the metrics or even possibly run malicious code in codevitals? Or is it scoped only under maintainers in this repo? Asking to be sure this is not a security risk. Please ignore me if I misunderstood the comment, or if this has already been thought of though!

Comment (youknowriad) [reply .github/workflows/performance.yml]: > Does this mean that anyone opening a PR could potentially skew the metrics or even possibly run malicious code in codevitals?

Anyone that has Gutenberg permissions, If I'm not wrong, fork PRs don't run CI unless approved.

**Pull request merged**